app/src/main/java/com/amaze/filemanager/utils/ScreenUtils.kt

```diff
@@ -22,9 +22,15 @@ package com.amaze.filemanager.utils
 
 import android.app.Activity
 import android.util.DisplayMetrics
+import java.lang.ref.WeakReference
 import kotlin.math.roundToInt
 
-class ScreenUtils(private val activity: Activity) {
+class ScreenUtils(act: Activity) {
+
+    private val _activity: WeakReference<Activity> = WeakReference(act)
+    private val activity: Activity
+        get() = _activity.get()!!
```

Comment:
Whats up with this?

Reply:
Saw leakage with this, so use WeakReference to keep it. Removing the accessor at line 31.

Reply:
This also has a non null asserted on get() that might lead to crash on garbage collection.

Reply:
Maybe just return `Activity?` in the getter, and use the `?.` everywhere the activity is being called in class.